broccoli-image-writer.js

```diff
@@ -73,19 +64,21 @@ class ImageResizer extends CachingWriter {
    * @returns {Array} an array of promise-functions
    */
   generateImages(file, sourcePath, destinationPath) {
-    let promises = [];
-    let gmImage = gm(path.join(sourcePath, file));
-    let imageInfos = Q.all([
-      Q.ninvoke(gmImage, 'format'),
-      Q.ninvoke(gmImage, 'color'),
-      Q.ninvoke(gmImage, 'size')
-    ]);
-    this.image_options.supportedWidths.forEach((width) => {
-      promises.push(() => {
-        return imageInfos.then((infos) => this.generateImage(file, sourcePath, destinationPath, width, infos));
+    let image = sharp(path.join(sourcePath, file));
+    return image.metadata()
+      .then((meta) => {
+        let promises = this.image_options.supportedWidths.map((width) => {
+          return this.generateImage(
+            file,
+            sourcePath,
+            destinationPath,
+            width,
+            meta
+          );
+        });
+
+        return Promise.all(promises);
```

Comment:
This line can be removed

Reply:
Sorry, shouldn't be removed. My bad.

Reply:
Um, no? Why? The other `return` above if from the `.map` callback...

Reply:
But this line will collect many promises and raise the limit of the above `parallelLimit`  execution.

Reply:
Seems valid. 

However I'm unsure if/how that worked before, because before an array of promise was passed to `async.parallelLimit()`. But when the promises are already created, the associated "tasks" have already started, without `async.parallelLimit` having a chance to control that. That's why the API docs are saying you should pass an array of promise-creating functions to it, so it can control when to create the promises. See https://github.com/dbushong/async-q#parallelLimit

Or do I miss something?

Reply:
@andreasschacht this should be fixed now! Sorry, was confused...